### ISSUE TITLE
Unable to extractalignedregion from filterdb output of linsearch alignments

### ISSUE THREAD

**Issue by manu-script** [soedinglab/MMseqs2#368]
## Expected Behavior
Extract aligned regions from alignmentDB output of `filterdb` command and write sequenceDB 

## Current Behavior
Segmentation fault with the error:  `Can not write to data file`

## Steps to Reproduce (for bugs)

`mmseqs linsearch test/S29DEC19 db/nt test/aln tmpdir --min-seq-id 0.6 -c 0.8 --cov-mode 2 --search-type 3`
`mmseqs filterdb test/aln test/tophitaln --extract-lines 1`
`mmseqs extractalignedregion test/S29DEC19 db/nt test/tophitaln test/tophitseq`

## MMseqs Output (for bugs)

```
mmseqs filterdb test/aln test/tophitaln --extract-lines 1
MMseqs Version:                 12.113e3
Filter expression
Filter column                   1
Column to take                  -1
Filter regex                    ^.*$
Positive filter                 true
Filter file
Beats first                     false
Mapping file
Trim to one column              false
Extract N lines                 1
Numerical comparison operator
Numerical comparison value      0
Sort entries                    0
Include identical seq. id.      false
join to DB
Threads                         40
Compressed                      0
Verbosity                       3

Filtering by extracting the first 1 lines
[=================================================================] 100.00% 26.17M 3s 458ms
Time for merging to tophitaln: 0h 0m 9s 953ms
Time for processing: 0h 0m 15s 822ms
```

```
mmseqs extractalignedregion test/S29DEC19 db/nt test/tophitaln test/tophitseq

MMseqs Version: 12.113e3
Compressed      0
Extract mode    2
Preload mode    0
Threads         40
Verbosity       3
 
Can not write to data file test/tophitseq.14                      ] 0.00% 1 eta -

```

## Context

I understand that `linsearch` is still under active development. But I am unable to extract aligned regions after filtering `linsearch` alignment output for top hits with `filterdb.` However, `convertalis` is able to produce blast formatted output from the `filterdb` output. Please let me know if you need any other information to understand what is the problem.

Thanks,
Manu

**Comment by manu-script** [comment]
> `mmseqs extractalignedregion test/S29DEC19 db/nt test/tophitaln test/tophitseq`
> 
> MMseqs Version: 12.113e3
> Compressed      0
> Extract mode    2
> Preload mode    0
> Threads         40
> Verbosity       3
> 
> Can not write to data file test/tophitseq.14                      ] 0.00% 1 eta -

I tried to reproduce the above error on a different target database to see if its something to do with the sequences and got the same error with the following additional error messages. Just thought reporting it might help to figure out what's wrong.

`free(): corrupted unsorted chunks`
`Aborted (core dumped)`

@milot-mirdita or @martin-steinegger do you have any idea what's going on? Any help is appreciated. 

Thanks,
Manu

**Comment by milot-mirdita** [comment]
Does this also happen when you call the normal search?

**Comment by manu-script** [comment]
No. I tried the normal search with the same query and target databases and I was able to extract aligned regions without any problem!
